File: demo-contract/scripts/deploy.js
const {ethers, upgrades} = require("hardhat");


async function main() {
    const SSVETH = await ethers.getContractFactory("SSVETH");
    // const Common = await ethers.getContractFactory("Common");
    // const KeysManager = await ethers.getContractFactory("KeysManager");
    const StakingPoolABI = await ethers.getContractFactory("StakingPool");
    // OPTIONAL you can change your SSV operators here
    const operator_ids = [1, 2, 9, 42]
    // TODO change whitelist to address you will making tx to staking pool and keysmanager, probably your current depolyer
    const whitelist = "0x0000536dbD99d918092249Ef4eDe4a69A35CccCa"
    // OPTIONAL change to receive eth rewards
    const withdrawal_creds = "0x0000536dbD99d918092249Ef4eDe4a69A35CccCa"
    const deposit_contract = "0xff50ed3d0ec03ac01d4c79aad74928bff48a7b2b"
    const ssv_network_contract = "0xb9e155e65B5c4D66df28Da8E9a0957f06F11Bc04"
    const ssv_token_address = "0x3a9f01091C446bdE031E39ea8354647AFef091E7";
    // const oracle_address = "0xfabb0ac9d68b0b445fb7357272ff202c5651694a"

    console.log("deploying ssvETH...");
    const ssvETH = await SSVETH.deploy();
    await ssvETH.deployed();
    // const ssvETH = await ssvETH.attach("0x7a2088a1bFc9d81c55368AE168C2C02570cB814F");
    console.log("ssvETH deployed to:", ssvETH.address)

    // const common = await Common.deploy();
    // console.log("Common contract deployed to:", common.address)

    console.log("deploying staking Pool...");
    const stakingPool = await StakingPoolABI.deploy(whitelist, deposit_contract, withdrawal_creds, ssv_network_contract, ssv_token_address, ssvETH.address, operator_ids);
    await stakingPool.deployed();
    console.log("staking pool deployed to:", stakingPool.address)
    // tx = await ssvETH.setMinter(stakingpool.address);
    // await tx.wait()



    // console.log("adding values in common contract")
    // tx = await common.changeStakingPool(stakingpool.address);
    // await tx.wait();
    // tx = await common.changeRoETH(roETH.address);
    // await tx.wait();

    // tx = await common.changeOracle(oracle_address);
    // await tx.wait();


    // console.log("added values in common contract");
    // console.log(await common.getAdmin());


}

main()
    .then(() => process.exit(0))
    .catch(error => {
        console.error(error);
        process.exit(1);
    });

File: demo-contract/scripts/deploy.py
#!/usr/bin/python3
import time

from brownie import *
from scripts.utils.helpers import *
import json


def main():

    deployer = get_deployer()

    # TODO OPTIONAL update these values
    whitelist = "0x49c5e038dc918F841589cE517fF975A4429eEb5c"
    withdrawal_creds = "0x44449d7cA8e3724cb9c9E30Ce49B286e275D79bf"
    operator_ids = [4, 9, 17, 76]

    deposit_contract = "0xff50ed3d0ec03ac01d4c79aad74928bff48a7b2b"
    ssv_network_contract = "0xAfdb141Dd99b5a101065f40e3D7636262dce65b3"
    ssv_token_address = "0x3a9f01091C446bdE031E39ea8354647AFef091E7"

    # ssvETH = "0xfcE8F661AbBf1417E6f73593B9cf779aF83501a9"
    # stakingPool = "0xFc6f35B3D7e7d6c5789bC2c3566d8b1E9E11752a"

    print("deploying ssvETH...")
    # ssvETH = SSVETH.deploy({'from': deployer})
    # print('ssvETH deployed to: ', ssvETH.address)

    print("deploying staking Pool...")
    stakingPool = StakingPool.deploy(whitelist, deposit_contract, withdrawal_creds,
                                     ssv_network_contract, ssv_token_address, operator_ids, {'from': deployer, 'gas_price': 875000000000})  # to change nonce  add "'nonce': 142" into the curly brackets
    print("staking pool deployed to:", stakingPool.address)
    time.sleep(3)

    print("trasferring minting ownership...")
    ssvETH = SSVETH.at(stakingPool.ssvETH())
    ssvETH.transferOwnership(stakingPool.address, {
                             'from': deployer, 'gas_price': 875000000000})

    print("testing staking...")
    stakingPool.stake({'value': 65 * 10 ** 18,
                      'from': deployer, 'gas_price': 875000000000})

    print("testing unStaking...")

    ssvETH.approve(stakingPool.address, 0.001 * 10 ** 18,
                   {'from': deployer, 'gas_price': 875000000000})
    stakingPool.unStake(
        0.001 * 10 ** 18, {'from': deployer, 'gas_price': 875000000000})

    with open('contrat_addresses.json', "r") as f:
        data = json.load(f)

    data["stakingPool_addr"] = stakingPool.address
    data["ssvETH_addr"] = ssvETH.address

    print("writing contract addresses into contrat_addresses.json")
    with open('contrat_addresses.json', 'w') as f:
        json.dump(data, f)
